### DISCUSSION TITLE
'Issued at' claim ('iat') must be an Integer representing the time that the assertion was issued

### DISCUSSION THREAD

**Discussion by NikosVlagoidis** [community/community#24856] [Enterprise]
<p>I have a Github App on Github Enterprise 2.17. When I try to create an access token for my App using the <a href="https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-a-github-app" rel="nofollow noopener">JWT</a>  for the first 4 or 5 seconds, it is rejected by the server with the error:</p>
<p><em>401 - ‘Issued at’ claim (‘iat’) must be an Integer representing the time that the assertion was issued // See:</em> <a href="https://developer.github.com/enterprise/2.17/v3" target="_blank" rel="nofollow noopener">https://developer.github.com/enterprise/2.17/v3</a><br>
After 4 seconds pass and several retries server accepts the JWT and returns an access token</p>
<p>I have a script demonstrating the behavior:</p>
<pre><code>require 'json'
require 'openssl'
require 'octokit'
require 'jwt'
require 'time'
require 'pp'
require 'pry'

Octokit.configure do |c|
  c.api_endpoint = 'https://mygithubserver.com/api/v3/'
end

GITHUB_APP_ID = 2
GITHUB_APP_PRIVATE_KEY_FILE = 'github.app.pem'.freeze
GITHUB_INSTALLATION_ID = 4

payload = {
  iat: Time.now.to_i,
  exp: Time.now.to_i + (9 * 60),
  iss: GITHUB_APP_ID
}

# Cryptographically sign the JWT
private_pem = File.read(GITHUB_APP_PRIVATE_KEY_FILE)
private_key = OpenSSL::PKey::RSA.new(private_pem)
jwt = JWT.encode(payload, private_key, 'RS256')

client = Octokit::Client.new(bearer_token: jwt)

begin
  installation_token = client.create_app_installation_access_token(GITHUB_INSTALLATION_ID, accept: 'application/vnd.github.machine-man-preview+json')[:token]
rescue Octokit::Unauthorized =&gt; e
  puts "ERROR: #{e}"
  sleep(1)
  retry
end
</code></pre>
<p>Is this the expected behavior? </p>

**Comment by NikosVlagoidis** [comment, marked as answer]
<p>Looks like the systems’ clock was out of sync! The error is not so descriptive. I think it should be fixed!</p>

**Reply by ayush-tiwari26** [reply]
But can you please explain more? How do I make sure the error does not appear. In my case the jwt gets randomly accepted sometimes, while gets rejected sometimes. How do I make sure the sync issue does not bother?
To sign, i use:

```js
const now = Math.floor(Date.now() / 1000) - 60;
    const payload = {
        iat: now,
        exp: now + (10 * 40),
        iss: appId
    };
    return await jwt.sign(payload, privateKey, { algorithm: 'RS256' });
```

**Comment by AndreaGriffiths11** [comment]
<p>Thanks for the update <a class="mention" href="/nikosvlagoidis">@nikosvlagoidis</a> </p>
